### PR TITLE
M3-6046: dynamically generate cURL and CLI commands

### PR DESCRIPTION
**Pull request by CodyFinn-Akamai**
## Description 📝

Adds functions to generate Linode CLI and cURL commands to create Linodes.
This is a work in progress to just get the basics down. I want to move this to a single function or an object that can generate both cURL and Linode CLI comands.

## How to test 🧪

`yarn test generate-cURL`

### CODE REVIEW

**Comment by cpathipa** [comment]
@CodyFinn-Akamai This should go to `feature/api-cli` branch

File: packages/manager/src/utilities/generate-cURL.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const headers = [
+  '-H "Content-Type: application/json"',
+  '-H "Authorization: Bearer $TOKEN"',
+].join('\n');
+
+const generateCurlCommand = (data: {}, path: string) => {
+  const command = `curl ${headers}\n-X POST -d ${JSON.stringify(
+    data
+  )}\n https://api.linode.com/v4/${path}`;
+  return command.trimStart().trimEnd();
```
Comment (carrillo-erik) [inline packages/manager/src/utilities/generate-cURL.ts:12]: I think you can simply use the `.trim()` method here instead. I believe it accomplishes the same result you are looking for.

File: packages/manager/src/utilities/generate-cli.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+export const generateCLICommand = (data: {}) => {
+  const dataForCLI = Object.entries(data).map(convertJSONFieldToCLIArg);
+  return `
+linode-cli linodes create
+${dataForCLI.join('\n')}
+  `
+    .trimStart()
+    .trimEnd();
```
Comment (carrillo-erik) [inline packages/manager/src/utilities/generate-cli.ts:18]: Same thing in this instance as my previous comment, the `.trim()` method returns a new string without the whitespaces from beginning and end of the original string.

**Review by carrillo-erik — CHANGES_REQUESTED** [review]

**Review by carrillo-erik — APPROVED** [review]

**Comment by cpathipa** [comment]
Hold on merging this PR until M3-6047 merges to the feature branch.

File: packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx [hunk 3]
```diff
@@ -62,38 +65,14 @@ const fireGAEvent = (label: string) => {
   });
 };
 const ApiAwarenessModal = (props: Props) => {
-  const { isOpen, onClose, route } = props;
+  const { isOpen, onClose, route, payLoad } = props;
 
   const classes = useStyles();
 
   // This harcoded values will be removed as part of following story that generated dynamic CURL and CLI commands.
```
Comment (cpathipa) [inline packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:72]: We can remove this comment.

**Review by cpathipa — APPROVED** [review]
LGTM!

**Pull request merged**